herepy/utils.py

```diff
@@ -18,16 +18,23 @@ def encode_parameters(parameters):
           parameters (dict):
             dictionary of query parameters to be converted.
         Returns:
-          A URL-encoded string in "key=value&key=value" form
+          A URL-encoded string in "key=value&key=value" form. If the parameter value is
+            a dict, the encoded string is converted to "main_key[sub_key]=sub_value".
         """
         if parameters is None:
             return None
         if not isinstance(parameters, dict):
             raise HEREError("`parameters` must be a dict.")
         else:
-            return urlencode(
-                dict((k, v) for k, v in parameters.items() if v is not None)
-            )
+            parameters = dict((k, v) for k, v in parameters.items() if v is not None)
+            res = dict()
+            for key, value in parameters.items():
+                if isinstance(value, dict):
+                    for key2, val2 in value.items():
+                        res[f"{key}[{key2}]"] =  val2
```

Comment:
Please format the code, I can see an extra space before `val2`. And also use better variable names for `key2` and `val2`.

Reply:
Done, thanks.